src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -181,6 +188,11 @@ public void onResultNotifiedOld(ProgrammingExerciseParticipation participation)
             // This needs to be done as some test cases might not have been executed.
             result = testCaseService.updateResultFromTestCases(result, programmingExercise, !isSolutionParticipation && !isTemplateParticipation);
             resultRepository.save(result);
+
+            // If the solution participation was updated, also trigger the template participation build.
+            if (isSolutionParticipation) {
```

Comment:
I think this if statement is not enough. We also need to check if the results comes from a commit into the test repository. Only then we want to the template build to be triggered by Artemis.

In case the instructor pushes code into the solution repository, we don't want the template to be triggered, because nothing has actually changed for it.

Reply:
It was implemented, but clear from the docs and method names.
It is checked in the trigger method: https://github.com/ls1intum/Artemis/pull/882/files#diff-7ed78ac5c411345e9fd436091876692bR225

The reason why I did it this way is that I want to avoid accessing `result.submission` from within the transactional method. If we change something e.g. in the BambooService, this could break due to an unproxy error.